docs/walkthroughs/path-routing-envoydeployment.md

```diff
@@ -0,0 +1,352 @@
+# **Path based routing using an EnvoyDeployment**
+
+In this walkthrough we will show how you can use the MARIN3R EnvoyDeployment custom resouce to deploy Envoy as a Kubernetes Deployment and dynamically configure it to proxy requests to a couple of different applications based on the path of the request. We will first configure it with just one upstream application and then add the second one to see how the config is updated without the need of restarting the Envoy proxies at any time.
+
+## **Preparation**
+
+You need to have MARIN3R operator installed in the cluster and a DiscoveryService within the namespace you will be using. Follow the [installation instructions](../../README.md#installation) to do so if you haven't already.
+
+## **Deploy two applications that we will use as upstreams to proxy requests to**
+
+We will be using `nginxdemos/hello:plain-text` to deploy our two upstream applications
+
+```bash
+cat <<'EOF' | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: upstream-a
+  labels:
+    app: upstream-a
+spec:
+  selector:
+    matchLabels:
+      app: upstream-a
+  template:
+    metadata:
+      labels:
+        app: upstream-a
+    spec:
+      containers:
+        - name: nginx
+          image: nginxdemos/nginx-hello:plain-text
+          ports:
+            - containerPort: 8080
+              name: http
+              protocol: TCP
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: upstream-a
+  labels:
+    app: upstream-a
+spec:
+  type: ClusterIP
+  clusterIP: None
+  ports:
+    - name: http
+      port: 8080
+      targetPort: http
+  selector:
+    app: upstream-a
+EOF
+```
+
+```bash
+cat <<'EOF' | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: upstream-b
+  labels:
+    app: upstream-b
+spec:
+  selector:
+    matchLabels:
+      app: upstream-b
+  template:
+    metadata:
+      labels:
+        app: upstream-b
+    spec:
+      containers:
+        - name: nginx
+          image: nginxdemos/nginx-hello:plain-text
+          ports:
+            - containerPort: 8080
+              name: http
+              protocol: TCP
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: upstream-b
+  labels:
+    app: upstream-b
+spec:
+  type: ClusterIP
+  clusterIP: None
+  ports:
+    - name: http
+      port: 8080
+      targetPort: http
+  selector:
+    app: upstream-b
+EOF
+```
+
+Now we will prepare the config to make an Envoy proxypass all requests with path matching `/a` to the Deployment we named as `upstream-a`. As usual, we will deploy this config as an EnvoyConfig.
```

Comment:
```suggestion
Now we will prepare the config to make an Envoy ProxyPass all requests with path matching `/a` to the Deployment we named as `upstream-a`. As usual, we will deploy this config as an EnvoyConfig.
```

Reply:
Why camel case here? proxypass is not a proper noun, more kind of a proxy technique. We could also use forward here.

Reply:
I just Googled it and saw that is mostly used as camelcase (probably due to the Apache directive), feel free to ignore this comment :p

Reply:
Maybe better to go with "forward" as the directive does not exist in Envoy.

```suggestion
Now we will prepare the config to make an Envoy forward all requests with path matching `/a` to the Deployment we named as `upstream-a`. As usual, we will deploy this config as an EnvoyConfig.
```